frontend/workflows/experimentation/src/list-experiments.tsx

```diff
@@ -1,49 +1,72 @@
 import React, { useState } from "react";
 import { useNavigate } from "react-router-dom";
 import type { clutch as IClutch } from "@clutch-sh/api";
-import type { BaseWorkflowProps } from "@clutch-sh/core";
 import { ButtonGroup, client, Error, Row, Table } from "@clutch-sh/core";
 import { Container } from "@material-ui/core";
 import styled from "styled-components";
 
-interface ExperimentationSpecificationDataProps {
+interface ExperimentationDataProps {
   experiment: IClutch.chaos.experimentation.v1.Experiment;
+  columns: string[];
+  experimentTypes: any;
 }
 
-const ExperimentSpecificationData: React.FC<ExperimentationSpecificationDataProps> = ({
+const ExperimentData: React.FC<ExperimentationDataProps> = ({
   experiment,
+  columns,
+  experimentTypes,
 }) => {
-  const specification = experiment.testSpecification;
-  const data = specification?.abort ? specification.abort : specification.latency;
-
-  const defaultData = [
-    data.clusterPair.downstreamCluster,
-    data.clusterPair.upstreamCluster,
-    data.percent,
-  ];
-  if (specification?.abort) {
-    defaultData.push((data as IClutch.chaos.experimentation.v1.AbortFault).httpStatus);
+  const types = experimentTypes || {};
+  if (!Object.prototype.hasOwnProperty.call(types, experiment.config["@type"])) {
```

Comment:
what is this checking for? could you please add a comment.

Reply:
added comment 👍